src/frontend/js/api/joanie/gen/services/ContractsService.ts

```diff
@@ -0,0 +1,159 @@
+/* generated using openapi-typescript-codegen -- do no edit */
+/* istanbul ignore file */
+/* tslint:disable */
+/* eslint-disable */
+import type { Contract } from '../models/Contract';
+import type { PaginatedContractList } from '../models/PaginatedContractList';
+
+import type { CancelablePromise } from '../core/CancelablePromise';
+import type { BaseHttpRequest } from '../core/BaseHttpRequest';
+
+export class ContractsService {
+
+  constructor(public readonly httpRequest: BaseHttpRequest) {}
+
+  /**
+   * Contract Viewset to list & retrieve contracts owned by the authenticated user.
+   *
+   * GET /api/contracts/
+   * Return list of all contracts owned by the logged-in user.
+   *
+   * GET /api/contracts/<contract_id>/
+   * Return a contract if one matches the provided id,
+   * and it is owned by the logged-in user.
+   *
+   * GET /api/contracts/<contract_id>/download/
+   * Return a contract in PDF format when it is signed on.
+   * @param courseId
+   * @param courseProductRelationId
+   * @param id primary key for the record as UUID
+   * @param organizationId
+   * @param page A page number within the paginated result set.
+   * @param pageSize Number of results to return per page.
+   * @param productId
+   * @param signatureState * `unsigned` - Unsigned
+   * * `half_signed` - Partially signed
+   * * `signed` - Signed
+   * @returns PaginatedContractList
+   * @throws ApiError
+   */
+  public contractsList(
+    courseId?: string,
+    courseProductRelationId?: string,
+    id?: Array<string>,
+    organizationId?: string,
+    page?: number,
+    pageSize?: number,
+    productId?: string,
+    signatureState?: 'half_signed' | 'signed' | 'unsigned',
```

Comment:
It would be nice to have :
```ŧs
enum SignatureStateEnum {
 half_signed,
 signed,
 unsigned
}
// ... 
signatureState?: SignatureStateEnum
```